cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/partitioned/DynamicPartitioningOutputCommitter.java

```diff
@@ -194,12 +196,30 @@ public void abortJob(JobContext context, JobStatus.State state) throws IOExcepti
     // We want to rollback these partitions.
     if (outputDataset != null) {
       try {
-        ((TransactionAware) outputDataset).rollbackTx();
-      } catch (Throwable t) {
-        LOG.warn("Attempt to rollback partitions failed", t);
+        try {
+          ((TransactionAware) outputDataset).rollbackTx();
+        } catch (Throwable t) {
+          LOG.warn("Attempt to rollback partitions failed", t);
+        }
+
+        // if this is non-null, then at least some paths have been created. We need to remove them
+        if (relativePaths != null) {
```

Comment:
It only removed the ones that have already been added to the PFS with addPartition(). But until then, the PFS does not even know of the existence of these files. Therefore we have to remove them here